frame/pools/src/lib.rs

```diff
@@ -0,0 +1,1135 @@
+
+			// This is the equivalent of `(current_points / 1) * new_funds`.
+			new_funds.saturating_mul(current_points)
+		},
+		(false, false) => {
+			// Equivalent to (current_points / current_balance) * new_funds
+			current_points
+				.saturating_mul(new_funds)
+				// We check for zero above
+				.div(current_balance)
+		},
+	}
+}
+
+// Calculate the balance of a pool to unbond as `(current_balance / current_points) *
+// delegator_points`. Returns zero if any of the inputs are zero.
+fn balance_to_unbond<T: Config>(
+	current_balance: BalanceOf<T>,
+	current_points: BalanceOf<T>,
+	delegator_points: BalanceOf<T>,
+) -> BalanceOf<T> {
+	if current_balance.is_zero() || current_points.is_zero() || delegator_points.is_zero() {
+		// There is nothing to unbond
+		return Zero::zero()
+	}
+
+	// Equivalent of (current_balance / current_points) * delegator_points
+	current_balance
+		.saturating_mul(delegator_points)
+		// We check for zero above
+		.div(current_points)
+}
+
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound)]
+#[cfg_attr(feature = "std", derive(Clone, PartialEq))]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Delegator<T: Config> {
+	pool: T::AccountId,
+	/// The quantity of points this delegator has in the bonded pool or in a sub pool if
+	/// `Self::unbonding_era` is some.
+	points: BalanceOf<T>,
+	/// The reward pools total earnings _ever_ the last time this delegator claimed a payout.
+	/// Assuming no massive burning events, we expect this value to always be below total issuance.
+	/// This value lines up with the `RewardPool::total_earnings` after a delegator claims a
+	/// payout.
+	reward_pool_total_earnings: BalanceOf<T>,
+	/// The era this delegator started unbonding at.
+	unbonding_era: Option<EraIndex>,
+}
+
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound)]
+#[cfg_attr(feature = "std", derive(Clone, PartialEq))]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct BondedPool<T: Config> {
+	points: BalanceOf<T>,
+}
+
+impl<T: Config> BondedPool<T> {
+	/// Get the amount of points to issue for some new funds that will be bonded in the pool.
+	fn points_to_issue(&self, stash: &T::AccountId, new_funds: BalanceOf<T>) -> BalanceOf<T> {
+		let bonded_balance = T::StakingInterface::bonded_balance(stash).unwrap_or(Zero::zero());
+		points_to_issue::<T>(bonded_balance, self.points, new_funds)
+	}
+
+	// Get the amount of balance to unbond from the pool based on a delegator's points of the pool.
+	fn balance_to_unbond(
+		&self,
+		stash: &T::AccountId,
+		delegator_points: BalanceOf<T>,
+	) -> BalanceOf<T> {
+		let bonded_balance = T::StakingInterface::bonded_balance(stash).unwrap_or(Zero::zero());
+		balance_to_unbond::<T>(bonded_balance, self.points, delegator_points)
+	}
+
+	// Check that the pool can accept a member with `new_funds`.
+	fn ok_to_join_with(
+		&self,
+		stash: &T::AccountId,
```

Comment:
again, if this `stash` is the account of the Pool, it should be known by the pool itself.

Reply:
this is the root issue:

Reply:
either way

```suggestion
		pool_account: &T::AccountId,
```

Reply:
4834c94fe21676e388579c4db95df893af8f133a